common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -31,8 +31,8 @@ public static long nextPowerOf2(long num) {
     return (highBit == num) ? num : highBit << 1;
   }
 
-  public static int roundNumberOfBytesToNearestWord(int numBytes) {
-    int remainder = numBytes & 0x07;  // This is equivalent to `numBytes % 8`
+  public static long roundNumberOfBytesToNearestWord(long numBytes) {
```

Comment:
I think we should create a new method instead of changing this method. The added int cast looks scaring.